storage/client_range_gc_test.go

```diff
@@ -41,6 +41,20 @@ func TestRangeGCQueueDropReplica(t *testing.T) {
 	// Increment the clock's timestamp to expire the leader lease.
 	mtc.manualClock.Increment(int64(storage.DefaultLeaderLeaseDuration) + 1)
 
+	// Make sure the range is not yet removed from the store.
```

Comment:
couldn't you try `shouldQueue` (or whichever function actually checks whether the range should be queued) directly?